Handles empty country and city values

valentinyanakiev opened the pull request:
Ensures that the geoLocation is not updated when both the country and city are empty.

Sets the longitude and latitude to 0 in this scenario to prevent incorrect data from being saved.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Bug Fixes**
	- Improved handling of empty country and city fields in location data to prevent invalid geolocation updates. When both fields are empty, the system now clears longitude and latitude values immediately.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

valentinyanakiev committed:
Handles empty country and city values

Ensures that the geoLocation is not updated when both the country and city are empty.

Sets the longitude and latitude to 0 in this scenario to prevent incorrect data from being saved.

bobbykolev commented on src/domain/common/location/location.service.ts:
```diff
@@ -115,6 +115,15 @@ export class LocationService {
+    if (location.country === '' && location.city === '') {
+      // If both country and city are empty, we cannot update the geoLocation.
+      // In the ideal case the data should not be updated on query; to be discussed.
+      location.geoLocation.longitude = 0;
```
I know that this could be handled on the client that will consume the coordinates, but:

> Setting latitude and longitude to 0 (i.e., location.geoLocation.latitude = 0; and location.geoLocation.longitude = 0;) places the location at the coordinate (0, 0) on the globe. This point is known as the Null Island, which is located in the Gulf of Guinea, off the west coast of Africa, where the Equator and Prime Meridian intersect.

valentinyanakiev committed:
Merge branch 'develop' into empty-location

valentinyanakiev committed:
Sets geoLocation to undefined when missing

Updates longitude and latitude to undefined when both country and city are empty.
This prevents incorrect coordinate values when location data is incomplete.

valentinyanakiev commented on src/domain/common/location/location.service.ts:
```diff
@@ -115,6 +115,15 @@ export class LocationService {
       return location.geoLocation;
     }
+
+    if (location.country === '' && location.city === '') {
```
why would I assign undefined if the value is undefined already?
